container_images/concourse-metrics/pkg/monitoring/requests.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitoring
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/golang/protobuf/ptypes/timestamp"
+	metricpb "google.golang.org/genproto/googleapis/api/metric"
+	"google.golang.org/genproto/googleapis/api/monitoredres"
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+)
+
+// JobResultArgs defines the required and optional arguments for building a new
+// job result request (in the form of a cloud monitoring time series request).
+type JobResultArgs struct {
```

Comment:
comment suggests some of these are optional, how should we know which ones?

i would also say it's somewhat uncommon to use struct-of-parameters in method signatures, though probably fine in this usage

Reply:
I followed the convention that Google's go client seems to be following. Optional params having the ability to be 'nil' by being a pointer rather than an explicit input value. Also, go readability suggests that we use a struct like this when there starts to be a lot of params: go/go-option-structure

Reply:
Also, I wish there was better support for explicitly marking optional/required args. If you know of a better way, please let me know.

Reply:
so the go clients are autogenerated and therefore not good examples of idiomatic go, sadly. optional params will only be nil if they are pointers, something you can also do with explicit function parameters. 

generally since go doesn't have polymorphism we just ape polymorphism with multiple signatures, return a modifiable struct, or if it's unwieldy we use a struct as you're doing now. it's just not very common to need a lot of things in one method like this :)

Reply:
oh and the bit about using a struct was really an aside. the feedback here is to document the optional params